aikau/src/main/resources/alfresco/layout/VerticalReveal.js

```diff
@@ -83,61 +92,94 @@ define(["alfresco/core/ProcessWidgets",
-         // Set a boolean flag to record when the child widgets get processed so that we only process them once...
-         this._processedWidgetsToReveal = false;
-         
-         if (this.toggleLabel == null)
-         {
-            domClass.add(this.toggleNode, "hidden");
+      postMixInProperties: function alfresco_layout_VerticalReveal__postMixInProperties() {
+         if (this.toggleLabel) {
+            this.toggleLabel = this.message(this.toggleLabel);
          }
-         this.alfSubscribe(this.subscriptionTopic, lang.hitch(this, this.onDisplayToggle));
+         this.inherited(arguments);
+      },
 
-         if (this.initiallyRevealed === true && this.widgets)
-         {
-            this.processWidgets(this.widgets, this.containerNode);
-            this._processedWidgetsToReveal = true;
-         }
+      /**
+       * Handle requests to toggle the contents visibility
+       *
+       * @instance
+       */
+      _onDisplayToggle: function alfresco_layout_VerticalReveal___onDisplayToggle() {
+         this._processWidgets().then(lang.hitch(this, function() {
+            var maxHeight = this.contentNode.style.maxHeight, // domStyle.get returns 0 instead of "none"
+               isExpanded = maxHeight === "none" || (typeof maxHeight === "number" && maxHeight > 0);
+            if (isExpanded) {
+               domClass.remove(this.contentNode, "content--has-transition");
+               domStyle.set(this.contentNode, "maxHeight", this.contentNode.scrollHeight + "px");
+               domClass.add(this.contentNode, "content--has-transition");
+               setTimeout(lang.hitch(this, function() {
+                  domStyle.set(this.contentNode, "maxHeight", 0);
+               }), 0);
+            } else {
+               domStyle.set(this.contentNode, "maxHeight", this.contentNode.scrollHeight + "px");
+               setTimeout(lang.hitch(this, function() {
+                  domStyle.set(this.contentNode, "maxHeight", "none");
+               }), 1000);
+            }
+         }));
       },
 
       /**
-       * 
+       * Process the widgets to be displayed
        *
        * @instance
+       * @returns {object} A promise, that will be resolved when the widgets have been processed
        */
-      onDisplayToggle: function alfresco_layout_VerticalReveal__onDisplayToggle() {
-         if (this._processedWidgetsToReveal === false && this.widgets)
-         {
-            this.processWidgets(this.widgets, this.containerNode);
-            this._processedWidgetsToReveal = true;
-         }
-         var maxHeight = domStyle.get(this.contentNode, "maxHeight"),
-            isExpanded = !isNaN(maxHeight) && maxHeight > 0;
-         if (isExpanded) {
-            domStyle.set(this.contentNode, "maxHeight", 0);
-         } else {
-            domStyle.set(this.contentNode, "maxHeight", this.contentNode.scrollHeight + "px");
+      _processWidgets: function alfresco_layout_VerticalReveal___processWidgets() {
```

Comment:
I'm not really sure what the benefit this approach is beyond simply including a promise. The code could surely be simpler if just calling processWidgets the first time and then the flag being set in allWidgetsProcessed. I'm not sure where the additional value is, especially as this then differs from the pattern used everywhere else.